# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/tools/reftest/manifest.sys.mjs
# Commit: beec388d4b0e
# Full Hash: beec388d4b0e6a50ac243ed86c0e3f1f48cbe744
# Author: Ashley Hale <ahale@mozilla.com>
# Date: 2024-06-25 09:36:02
# Regressor Bug: 1896503
# File Overlap Count: 1
# Description:
#   Bug 1896503 - Implement FilterInstance code to send SVG filter graph to WebRender r=mstange,gw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D174208
# ==============================================================================

diff -r 2147e0051e23 -r beec388d4b0e layout/tools/reftest/manifest.sys.mjs
--- a/layout/tools/reftest/manifest.sys.mjs	Mon Jun 24 23:40:38 2024 +0000
+++ b/layout/tools/reftest/manifest.sys.mjs	Mon Jun 24 23:45:05 2024 +0000
@@ -648,6 +648,36 @@
   sandbox.swgl = g.windowUtils.layerManagerType.startsWith(
     "WebRender (Software"
   );
+  // These detect if each SVG filter primitive is enabled in WebRender
+  sandbox.gfxSVGFE =
+    Services.prefs.getBoolPref("gfx.webrender.svg-filter-effects") &&
+    !g.useDrawSnapshot;
+  sandbox.gfxSVGFEColorMatrix =
+    Services.prefs.getBoolPref(
+      "gfx.webrender.svg-filter-effects.fecolormatrix"
+    ) && sandbox.gfxSVGFE;
+  sandbox.gfxSVGFEComponentTransfer =
+    Services.prefs.getBoolPref(
+      "gfx.webrender.svg-filter-effects.fecomponenttransfer"
+    ) && sandbox.gfxSVGFE;
+  sandbox.gfxSVGFEComposite =
+    Services.prefs.getBoolPref(
+      "gfx.webrender.svg-filter-effects.fecomposite"
+    ) && sandbox.gfxSVGFE;
+  sandbox.gfxSVGFEDropShadow =
+    Services.prefs.getBoolPref(
+      "gfx.webrender.svg-filter-effects.fedropshadow"
+    ) && sandbox.gfxSVGFE;
+  sandbox.gfxSVGFEFlood =
+    Services.prefs.getBoolPref("gfx.webrender.svg-filter-effects.feflood") &&
+    sandbox.gfxSVGFE;
+  sandbox.gfxSVGFEGaussianBlur =
+    Services.prefs.getBoolPref(
+      "gfx.webrender.svg-filter-effects.fegaussianblur"
+    ) && sandbox.gfxSVGFE;
+  sandbox.gfxSVGFEOffset =
+    Services.prefs.getBoolPref("gfx.webrender.svg-filter-effects.feoffset") &&
+    sandbox.gfxSVGFE;
 
   // Use this to annotate reftests that fail in drawSnapshot, but
   // the reason hasn't been investigated (or fixed) yet.